src/main/java/com/synopsys/integration/jenkins/scan/bridge/BridgeDownload.java

```diff
@@ -0,0 +1,87 @@
+package com.synopsys.integration.jenkins.scan.bridge;
+
+import com.synopsys.integration.jenkins.scan.global.ApplicationConstants;
+
+import hudson.FilePath;
+import hudson.model.TaskListener;
+
+import java.io.File;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BridgeDownload {
+    private final TaskListener listener;
+
+    public BridgeDownload(TaskListener listener) {
+        this.listener = listener;
+    }
+
+    public FilePath downloadSynopsysBridge(String bridgeVersion, String bridgeDownloadUrl) {
+        String bridgeUrl = null;
+
+        if (bridgeVersion.equals(ApplicationConstants.SYNOPSYS_BRIDGE_LATEST_VERSION)) {
```

Comment:
As per this logic, we are giving precedence to version. But as per https://sig-confluence.internal.synopsys.com/display/PRODM/Airgap+Support we should give precedence to Bridge download url. So let's give precedence to url

Reply:
Here, We aren't giving any precedence to anything. We are simply constructing the 'Download Url', from which bridge will be downloaded. 
By default `bridgeDownloadUrl = https://sig-repo.synopsys.com/artifactory/bds-integrations-release/com/synopsys/integration/synopsys-bridge/`, which contains list of directories of released `synopsys-bridge` executable.

We are just adding required version to the `bridgeDownloadUrl` to get to the desired released version.

Makes sense?

Reply:
By default synopsys artifactory url will be selected. But if user provides both **download url and bridge version**, then specified version will be downloaded from the specified url.

https://github.com/synopsys-sig/synopsys-security-scan-plugin/blob/dev/src/main/java/com/synopsys/integration/jenkins/scan/service/BridgeDownloadParametersService.java#L57

Reply:
As per our discussion, I will implement the caching issue in a separate branch with the updated requirement.